### PR TITLE
Pull out RPC API changes from #995.

### PR DESCRIPTION
**Pull request by Martin2112**
Adds SignedLogRoot to all the responses. This enables CTFE to be modified before server behaviour changes.

### CODE REVIEW

File: trillian_log_api.proto [hunk 1]
```diff
@@ -85,13 +85,16 @@ service TrillianLog {
     // Returns the total number of leaves that have been integrated into the
     // given tree. Corresponds to the ReadOnlyLogTreeTX.GetSequencedLeafCount
     // storage interface.
+    // DO NOT USE - FOR DEBUGGING/TEST ONLY (FIXME: remove or move to a debugging interface)
```
Comment (daviddrysdale) [inline trillian_log_api.proto]: Could we have the FIXME as a TODO please

**Comment by daviddrysdale** [comment]
Also, could we make the commit description self-contained please?  E.g. something like:

"API: add SignedLogRoot to all Get* responses"

File: trillian_log_api.proto (hunk 1 above)
Comment (Martin2112) [reply trillian_log_api.proto]: I'll remove the FIXME part for the moment as it's in use by the MySQL tests so we can't just delete it.

**Comment by Martin2112** [comment]
Commit renamed.

**Pull request merged**